src/freenas/etc/rc.conf.local

```diff
@@ -622,7 +622,7 @@ _gen_conf()
 	## Daemons
 	_bool_svcconfig netatalk_enable afp
 	_bool_svcconfig proftpd_enable ftp
-	if [ ${failover_licensed} -ne 1 ]; then
+	if [ "${failover_status}" != "BACKUP" ]; then
```

Comment:
failover_status cannot be relied on, its not guaranteed to be valid by the time the cache is generated.

Reply:
@william-gr If we can't trust `failover_status` then there are a couple other places in that script that use that same check. Do we need to worry about those as well?

Reply:
Possibly

Reply:
So I'm going to be honest here. That config file is potentially breaking many services. For example, if any services rc.conf settings are dictated by the `if [${failover_licensed} -ne 1]` check then we're not setting the rc.conf.local file correctly for TrueNAS HA systems. We're starting the services without issues because carp-state change hook doesn't have any notion of that file, however, if we use the `service` command to manage any services we're not setting rc.conf settings appropriately. I'm not sure where to really go with this one without making logic changes in other portions of the code which would probably be invasive. Any hints on how you'd like this done @william-gr or do we revert the changes in commit: 41646cc5676a645ae4364b682c6dcaa2a3e3348f ?

Reply:
Further investigation shows that this is also the reason why the `vfs.nfsd.server_max_nfsvers` sysctl isn't being set back to `3` when we restart nfsd daemon via the `service` framework because the rc script is automatically changing that value back because we're not setting `nfsv4_server_enable="YES"` in /tmp/rc.conf.local on TrueNAS HA. The ticket for that issue is here: https://jira.ixsystems.com/browse/NAS-103502 So it seems like either we make invasive changes somewhere else or we revert commit:41646cc5676a645ae4364b682c6dcaa2a3e3348f

Reply:
I'm inclined to revert the commit at this point because that fix was only to fix some cosmetic console error messages when TrueNAS passive controller booted up.

Reply:
```suggestion
	if [ "${failover_status}" = "MASTER" -o "${failover_status}" = "SINGLE" ]; then
```
If it's written this way, are there any other conditions that should be covered? I assume the cache is regenerated and services are started or stopped accordingly when the failover status changes.

Reply:
Scratch that, $failover_status isn't SINGLE for FreeNAS, so it should be more like
```suggestion
	if [ "${failover_status}" = "MASTER" \
	    -o "${failover_status}" = "SINGLE" \
	    -o ${failover_licensed} -ne 1 ]; then
```